dev/app.go

```diff
@@ -639,3 +593,81 @@ func (a *AppPool) Purge() {
 
 	a.Events.Add("apps_purged")
 }
+
+const pumaShellScriptTemplate = `exec %s -c '
+cd %s
+
+if test -e ~/.powconfig && [ "$PUMADEV_SOURCE_POWCONFIG" != "0" ]; then
+  source ~/.powconfig
+fi
+
+if test -e ~/.pumaenv && [ "$PUMADEV_SOURCE_HOME_PUMAENV" != "0" ]; then
+  source ~/.pumaenv
+fi
+
+if test -e .env && [ "$PUMADEV_SOURCE_ENV" != "0" ]; then
+  source .env
+fi
+
+if test -e .powrc && [ "$PUMADEV_SOURCE_POWRC" != "0" ]; then
+  source .powrc
+fi
+
+if test -e .powenv && [ "$PUMADEV_SOURCE_POWENV" != "0" ]; then
+  source .powenv
+fi
+
+if test -e .pumaenv && [ "$PUMADEV_SOURCE_PUMAENV" != "0" ]; then
+  source .pumaenv
+fi
+
+if test -e Gemfile && bundle exec puma -V &>/dev/null; then
+  exec bundle exec puma %s
+fi
+
+exec puma %s
+'` // <-- don't forget this closing quote
+
+func BuildPumaCommand(appName string, socketPath string, appDir string) (*exec.Cmd, error) {
+
+	osMapEnv := GetMapEnviron()
+
+	// This environment will be fed into the puma exec shell.
+	// However, we _also_ source each supported env file inside the shell to allow for
+	// custom environment grooming.
+	var mapEnv map[string]string
+	mapEnv, err := LoadEnv(osMapEnv, appDir)
+	if err != nil {
+		fmt.Printf("! Falling back to default environment. %v\n", err.Error())
+		mapEnv = osMapEnv
+	}
```

Comment:
this way, we'll load up your env in go if we can, and fallback to the old behavior if we can't. This also serves as a decent spot to stuff a deprecation warning into the logs, if we ever want rip out the shell sourcing. And, in that case, prob want to add it to the post-install message in homebrew.

One downside here is that if you have `WORKERS` specified in a file that also has shellcode, it won't get fed to the commandline correctly. But, that's easily rectified with a `config/puma.rb` anyway.

@cjlarose wdyt?

Reply:
Replied at the top level thread

Reply:
Whoa, did you ever 😂 Thanks for such a detailed and well-reasoned response!